### PR TITLE
HSEARCH-4167 Improve parameter handling in named predicates

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4167

### CODE REVIEW

File: integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/NamedPredicateBaseIT.java [hunk 1]
```diff
@@ -150,6 +150,18 @@ public void unknownPredicate_valueField() {
...
+		assertThatThrownBy( () -> {
+			index.query()
+					.where( f -> f.named( "match-both-fields" )
+							.param( "value1", WORD_1 ) );
+		} )
+				.isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						"Param with name 'value2' has not been defined for the named predicate 'match-both-fields'." );
+	}
+
```
Comment (yrodiere) [inline integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/NamedPredicateBaseIT.java:164]: Wouldn't it make more sense to put this test next to `org.hibernate.search.integrationtest.backend.tck.search.predicate.NamedPredicateProviderContextIT#param_nullName`?

**Review by yrodiere — COMMENTED** [review]
Thanks. Just a few comments.

**Comment by fax4ever** [comment]
Thanks for the review @yrodiere.
I applied all the changes.

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**